uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/RouteContextSupportTest.java

```diff
@@ -0,0 +1,156 @@
+		
+		/*
+		 * Change content in 'Source' tab for 'route-context.xml'.
+		 */
+		CamelEditor.switchTab("Source");
+		AbstractWait.sleep(TimePeriod.getCustom(3));
+		SourceEditor editor_route = new SourceEditor();
+		editor_route.selectText(0, 999999);		
+		editor_route.insertText("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+				+ "<beans xmlns=\"http://www.springframework.org/schema/beans\"\n"
+				+ "    xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:schemaLocation=\"    http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-4.2.xsd    http://camel.apache.org/schema/spring http://camel.apache.org/schema/spring/camel-spring.xsd\">\n"
+				+ "    <routeContext id=\"ExternalRoutes\" xmlns=\"http://camel.apache.org/schema/spring\">\n"
+				+ "        <route id=\"cbr-route\">\n"
+				+ "            <from id=\"_from1\" uri=\"file:work/cbr/input\"/>\n"
+				+ "            <log id=\"_log1\" message=\"Receiving order ${file:name}\"/>\n"
+				+ "            <choice id=\"_choice1\">\n"
+				+ "                <when id=\"_when1\">\n"
+				+ "                    <xpath id=\"_xpath1\">/order/customer/country = 'UK'</xpath>\n"
+				+ "                    <log id=\"_log2\" message=\"Sending order ${file:name} to the UK\"/>\n"
+				+ "                    <to id=\"_to1\" uri=\"file:work/cbr/output/uk\"/>\n"
+				+ "                </when>\n"
+				+ "                <when id=\"_when2\">\n"
+				+ "                    <xpath id=\"_xpath2\">/order/customer/country = 'US'</xpath>\n"
+				+ "                    <log id=\"_log3\" message=\"Sending order ${file:name} to the US\"/>\n"
+				+ "                    <to id=\"_to2\" uri=\"file:work/cbr/output/us\"/>\n"
+				+ "                </when>\n"
+				+ "                <otherwise id=\"_otherwise1\">\n"
+				+ "                    <log id=\"_log4\" message=\"Sending order ${file:name} to another country\"/>\n"
+				+ "                    <to id=\"_to3\" uri=\"file:work/cbr/output/others\"/>\n"
+				+ "                </otherwise>\n"
+				+ "            </choice>\n"
+				+ "            <log id=\"_log5\" message=\"Done processing ${file:name}\"/>\n"
+				+ "        </route>\n"
+				+ "    </routeContext>\n"
+				+ "</beans>");			
+		editor_route.save();
+				
+		
+		/*
+		 * Check if design tab contains <routeContext>
+		 */
+		CamelEditor.switchTab("Design");
+		Collection<TreeItem> tree_items = new ContentOutline().outlineElements();
+		String rootOutlineElement = tree_items.stream().findFirst().get().getText();	
+		String rootOutlineElementSplit [] = rootOutlineElement.split(" ", 2);	
+		if(!rootOutlineElementSplit[0].equals("RouteContext")) {
+			fail("RouteContext not found");
+		}
+		
+		AbstractWait.sleep(TimePeriod.getCustom(3));
+		
+		/*
+		 * Run as Local Camel Context (without tests).
+		 */
+		//Shell workbenchShell = new WorkbenchShell();
+		new CamelProject(PROJECT_NAME).runCamelContextWithoutTests(ROUTES_NAME);
+		try {
```

Comment:
try/catch block with `fail` is not needed. It will fail due to timeout anyway and that is enough.